The trait `CacheHttp` is not implemented for `Cache`

**Affected branch:** `Next`

As the title says, during compilation I get the following error:
> error[E0277]: the trait bound `Cache: CacheHttp` is not satisfied
```
let test = msg.channel(*ctx.cache).await;
                       ^^^^^^^^^^ the trait `CacheHttp` is not implemented for `Cache`
```

This appears to be either a bug or mistake in the documentation itself, since according to the documentation:
> The types Context, Cache, and Http implement this trait and thus passing these to functions expecting impl CacheHttp is possible.

**Related code fragment:**
```rust
async fn message(&self, ctx: Context, msg: Message) {
     let test = msg.channel(*ctx.cache).await;
}
```
The same issue happens when I try to pass `ctx.cache` without dereferencing it, as `Arc<Cache>`, which I assume is also a bug, since `Arc<Http>` can be passed over as atomic reference just fine. 

For the sake of clarity, I will mention that using `&ctx` alone in the same use case is working just fine, which would make this issue rather low priority.

**The following code works as intended:**
```rust
async fn message(&self, ctx: Context, msg: Message) {
     let test = msg.channel(&ctx).await;
}
```

--- a/src/http/mod.rs
+++ b/src/http/mod.rs
@@ -52,15 +52,17 @@ use crate::CacheAndHttp;
 /// This trait will be required by functions that need [`Http`] and can
 /// optionally use a [`Cache`] to potentially avoid REST-requests.
 ///
-/// The types [`Context`], [`Cache`], and [`Http`] implement this trait
-/// and thus passing these to functions expecting `impl CacheHttp` is possible.
+/// The types [`Context`] and [`Http`] implement this trait
+/// and thus passing these to functions expecting `impl CacheHttp` is possible. For the full list
+/// of implementations, see the Implementors and Implementations on Foreign Types section in the
+/// generated docs.
 ///
 /// In a situation where you have the `cache`-feature enabled but you do not
 /// pass a cache, the function will behave as if no `cache`-feature is active.
 ///
 /// If you are calling a function that expects `impl CacheHttp` as argument
 /// and you wish to utilise the `cache`-feature but you got no access to a
-/// [`Context`], you can pass a tuple of `(CacheRwLock, Http)`.
+/// [`Context`], you can pass a tuple of `(&Arc<Cache>, &Http)`.
 pub trait CacheHttp: Send + Sync {
     fn http(&self) -> &Http;
 

